src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotJVMCIBackendFactory.java

```diff
@@ -80,4 +81,58 @@ static <CPUFeatureType extends Enum<CPUFeatureType>> EnumSet<CPUFeatureType> con
         }
         return outFeatures;
     }
+
+    /**
+     * Converts CPU features bit map into enum constants.
+     *
+     * @param <CPUFeatureType> CPU feature enum type
+     * @param enumType the class of {@code CPUFeatureType}
+     * @param constants VM constants. Each entry whose key starts with {@code "VM_Version::CPU_"}
+     *            specifies a CPU feature and its value is a mask for a bit in {@code features}
+     * @param featuresBitMapAddress pointer to {@code VM_Features::_features_bitmap} field of {@code VM_Version::_features}
+     * @param featuresBitMapSize size of feature bit map in bytes
+     * @param renaming maps from VM feature names to enum constant names where the two differ
+     * @throws IllegalArgumentException if any VM CPU feature constant cannot be converted to an
+     *             enum value
+     * @return the set of converted values
+     */
+    static <CPUFeatureType extends Enum<CPUFeatureType>> EnumSet<CPUFeatureType> convertFeatures(
+                    Class<CPUFeatureType> enumType,
+                    Map<String, Long> constants,
+                    long featuresBitMapAddress,
+                    long featuresBitMapSize,
+                    Map<String, String> renaming) {
+        EnumSet<CPUFeatureType> outFeatures = EnumSet.noneOf(enumType);
+        List<String> missing = new ArrayList<>();
+
+        for (Entry<String, Long> e : constants.entrySet()) {
+            String key = e.getKey();
+            long bitIndex = e.getValue();
+            if (key.startsWith("VM_Version::CPU_")) {
+                String name = key.substring("VM_Version::CPU_".length());
+                try {
+                    final long featuresElementShiftCount = 6; // log (# of bits per long)
+                    final long featuresElementMask = (1L << featuresElementShiftCount) - 1;
+
+                    CPUFeatureType feature = Enum.valueOf(enumType, renaming.getOrDefault(name, name));
+
+                    long featureIndex = bitIndex >>> featuresElementShiftCount;
+                    long featureBitMask = 1L << (bitIndex & featuresElementMask);
+                    assert featureIndex < featuresBitMapSize;
```

Comment:
`featuresBitMapSize` is size in bytes while `featureIndex` is index to long array

Reply:
Good catch, Yudi.